Can't compile with time component in version 1.9 (1.8.x working)

<!-- Thanks for reporting a bug for this project. READ THIS FIRST:
- Please make sure to submit issues in the right GitHub repository, if unsure just post it here:
  - esphomeyaml [here] - This is mostly for reporting bugs when compiling and when you get a long stack trace while compiling or if a configuration fails to validate.
  - esphomelib [https://github.com/OttoWinter/esphomelib] - Report bugs there if the ESP is crashing or a feature is not working as expected.
  - esphomedocs [https://github.com/OttoWinter/esphomedocs] - Report bugs there if the documentation is wrong/outdated.
- Provide as many details as possible. Paste logs, configuration sample and code into the backticks (```). Do not delete any text from this template!
-->

**Operating environment (Hass.io/Docker/pip/etc.):**

pip

**ESP (ESP32/ESP8266/Board/Sonoff):**

esp8285 (esp m3)

**Affected component:**

not sure

**Description of problem:**

Can't compile (and upload) my sketch anymore in version 1.9. Did touch/change my yaml. Compiling with version 1.8.2 was working as expected.

**Problem-relevant YAML-configuration entries:**
```
i2c:
  sda: 0
  scl: 2
  scan: False

binary_sensor:
  - platform: status
    name: "ESPM3 Status"
    id: status

sensor:
  - platform: wifi_signal
    name: "ESPM3 WiFi Signal"
    update_interval: 300s
    id: signal

  - platform: bmp085
    temperature:
      name: "ESPM3 Temperature"
      id: temperature
    pressure:
      name: "ESPM3 Pressure"
      id: pressure
    update_interval: 300s

font:
  - file: "five.ttf"
    id: five8
    size: 8

time:
  - platform: sntp
    id: time
    servers:
      - 0.pool.ntp.org

display:
  - platform: ssd1306_i2c
    model: "SSD1306 128x64"
    address: 0x3C
    lambda:  |-
      it.rectangle(1, 1, 126, 32);
      it.printf(20, 10, id(five8), "%.1f°C", id(temperature).value);
      it.printf(84, 10, id(five8), "%.0fhPa", id(pressure).value);
      it.printf(5, 26, id(five8), "Temperature Pressure");
      it.strftime(2, 35, id(five8), "Time: %H:%M", id(time).now());
      it.strftime(2, 45, id(five8), "Date: %d.%m.%Y", id(time).now());
      it.printf(2, 60, id(five8), "Wifi: %s", id(status).value ? "Online," : "Offline");
      it.printf(80, 60, id(five8), "%.1fdb", id(signal).value);

```

**Traceback (if applicable):**
```
Compiling .pioenvs/espm3_solar/lib55f/ESP8266WiFi/ESP8266WiFiAP.cpp.o
src/main.cpp: In function 'void setup()':
src/main.cpp:110:37: error: no matching function for call to 'esphomelib::time::SNTPComponent::set_servers(const char [15])'
time->set_servers("0.pool.ntp.org");
^
src/main.cpp:110:37: note: candidate is:
In file included from .piolibdeps/esphomelib/src/esphomelib/application.h:128:0,
from src/main.cpp:2:
.piolibdeps/esphomelib/src/esphomelib/time/sntp_component.h:28:8: note: void esphomelib::time::SNTPComponent::set_servers(const string&, const string&, const string&)
void set_servers(const std::string &server_1,
^
.piolibdeps/esphomelib/src/esphomelib/time/sntp_component.h:28:8: note:   candidate expects 3 arguments, 1 provided
src/main.cpp: In lambda function:
src/main.cpp:117:56: warning: 'esphomelib::sensor::Sensor::<anonymous union>::value' is deprecated (declared at .piolibdeps/esphomelib/src/esphomelib/sensor/sensor.h:129): .value is deprecated. Please use .state instead [-Wdeprecated-declarations]
it.printf(20, 10, five8, "%.1f°C", temperature->value);
^
src/main.cpp:117:56: warning: 'esphomelib::sensor::Sensor::<anonymous union>::value' is deprecated (declared at .piolibdeps/esphomelib/src/esphomelib/sensor/sensor.h:129): .value is deprecated. Please use .state instead [-Wdeprecated-declarations]
src/main.cpp:118:53: warning: 'esphomelib::sensor::Sensor::<anonymous union>::value' is deprecated (declared at .piolibdeps/esphomelib/src/esphomelib/sensor/sensor.h:129): .value is deprecated. Please use .state instead [-Wdeprecated-declarations]
it.printf(84, 10, five8, "%.0fhPa", pressure->value);
^
src/main.cpp:118:53: warning: 'esphomelib::sensor::Sensor::<anonymous union>::value' is deprecated (declared at .piolibdeps/esphomelib/src/esphomelib/sensor/sensor.h:129): .value is deprecated. Please use .state instead [-Wdeprecated-declarations]
src/main.cpp:122:51: warning: 'esphomelib::binary_sensor::BinarySensor::<anonymous union>::value' is deprecated (declared at .piolibdeps/esphomelib/src/esphomelib/binary_sensor/binary_sensor.h:61): .value is deprecated, please use .state instead [-Wdeprecated-declarations]
it.printf(2, 60, five8, "Wifi: %s", status->value ? "Online," : "Offline");
^
src/main.cpp:122:51: warning: 'esphomelib::binary_sensor::BinarySensor::<anonymous union>::value' is deprecated (declared at .piolibdeps/esphomelib/src/esphomelib/binary_sensor/binary_sensor.h:61): .value is deprecated, please use .state instead [-Wdeprecated-declarations]
src/main.cpp:123:50: warning: 'esphomelib::sensor::Sensor::<anonymous union>::value' is deprecated (declared at .piolibdeps/esphomelib/src/esphomelib/sensor/sensor.h:129): .value is deprecated. Please use .state instead [-Wdeprecated-declarations]
it.printf(80, 60, five8, "%.1fdb", signal->value);
^
src/main.cpp:123:50: warning: 'esphomelib::sensor::Sensor::<anonymous union>::value' is deprecated (declared at .piolibdeps/esphomelib/src/esphomelib/sensor/sensor.h:129): .value is deprecated. Please use .state instead [-Wdeprecated-declarations]
Compiling .pioenvs/espm3_solar/lib55f/ESP8266WiFi/ESP8266WiFiGeneric.cpp.o
*** [.pioenvs/espm3_solar/src/main.cpp.o] Error 1

```

**Additional information:**

I did a extra downgrade back to 1.8.2 and can confirm that compiling the exact yaml works again

--- a/esphomeyaml/components/light/__init__.py
+++ b/esphomeyaml/components/light/__init__.py
@@ -1,18 +1,16 @@
 import voluptuous as vol
 
-from esphomeyaml.automation import maybe_simple_id, ACTION_REGISTRY
+from esphomeyaml.automation import ACTION_REGISTRY, maybe_simple_id
 from esphomeyaml.components import mqtt
 import esphomeyaml.config_validation as cv
 from esphomeyaml.const import CONF_ALPHA, CONF_BLUE, CONF_BRIGHTNESS, CONF_COLORS, \
-    CONF_DEFAULT_TRANSITION_LENGTH, CONF_DURATION, CONF_EFFECTS, CONF_EFFECT_ID, \
-    CONF_GAMMA_CORRECT, CONF_GREEN, CONF_ID, CONF_INTERNAL, CONF_LAMBDA, CONF_MQTT_ID, CONF_NAME, \
-    CONF_NUM_LEDS, CONF_RANDOM, CONF_RED, CONF_SPEED, CONF_STATE, CONF_TRANSITION_LENGTH, \
-    CONF_UPDATE_INTERVAL, CONF_WHITE, CONF_WIDTH, CONF_FLASH_LENGTH, CONF_COLOR_TEMPERATURE, \
-    CONF_EFFECT
-from esphomeyaml.helpers import Application, ArrayInitializer, Pvariable, RawExpression, \
-    StructInitializer, add, add_job, esphomelib_ns, process_lambda, setup_mqtt_component, \
-    get_variable, TemplateArguments, templatable, uint32, float_, std_string, Nameable, Component, \
-    Action
+    CONF_COLOR_TEMPERATURE, CONF_DEFAULT_TRANSITION_LENGTH, CONF_DURATION, CONF_EFFECT, \
+    CONF_EFFECTS, CONF_EFFECT_ID, CONF_FLASH_LENGTH, CONF_GAMMA_CORRECT, CONF_GREEN, CONF_ID, \
+    CONF_INTERNAL, CONF_LAMBDA, CONF_MQTT_ID, CONF_NAME, CONF_NUM_LEDS, CONF_RANDOM, CONF_RED, \
+    CONF_SPEED, CONF_STATE, CONF_TRANSITION_LENGTH, CONF_UPDATE_INTERVAL, CONF_WHITE, CONF_WIDTH
+from esphomeyaml.helpers import Action, Application, ArrayInitializer, Component, Nameable, \
+    Pvariable, StructInitializer, TemplateArguments, add, add_job, esphomelib_ns, float_, \
+    get_variable, process_lambda, setup_mqtt_component, std_string, templatable, uint32
 
 PLATFORM_SCHEMA = cv.PLATFORM_SCHEMA.extend({
 
@@ -32,7 +30,6 @@ TurnOffAction = light_ns.class_('TurnOffAction', Action)
 TurnOnAction = light_ns.class_('TurnOnAction', Action)
 
 LightColorValues = light_ns.class_('LightColorValues')
-
 
 MQTTJSONLightComponent = light_ns.class_('MQTTJSONLightComponent', mqtt.MQTTComponent)
 
@@ -353,7 +350,6 @@ def setup_light(light_obj, mqtt_obj, config):
 
 
 BUILD_FLAGS = '-DUSE_LIGHT'
-
 
 CONF_LIGHT_TOGGLE = 'light.toggle'
 LIGHT_TOGGLE_ACTION_SCHEMA = maybe_simple_id({

--- a/esphomeyaml/components/time/sntp.py
+++ b/esphomeyaml/components/time/sntp.py
@@ -9,7 +9,7 @@ SNTPComponent = time_.time_ns.class_('SNTPComponent', time_.RealTimeClockCompone
 
 PLATFORM_SCHEMA = time_.TIME_PLATFORM_SCHEMA.extend({
     cv.GenerateID(): cv.declare_variable_id(SNTPComponent),
-    vol.Optional(CONF_SERVERS): vol.All(cv.ensure_list, [cv.string], vol.Length(max=3)),
+    vol.Optional(CONF_SERVERS): vol.All(cv.ensure_list, [cv.string], vol.Length(min=1, max=3)),
     vol.Optional(CONF_LAMBDA): cv.lambda_,
 }).extend(cv.COMPONENT_SCHEMA.schema)
 
@@ -18,7 +18,9 @@ def to_code(config):
     rhs = App.make_sntp_component()
     sntp = Pvariable(config[CONF_ID], rhs)
     if CONF_SERVERS in config:
-        add(sntp.set_servers(*config[CONF_SERVERS]))
+        servers = config[CONF_SERVERS]
+        servers += [''] * (3 - len(servers))
+        add(sntp.set_servers(*servers))
 
     time_.setup_time(sntp, config)
     setup_component(sntp, config)

